sklearn/linear_model/_theil_sen.py

```diff
@@ -322,6 +322,18 @@ class TheilSenRegressor(RegressorMixin, LinearModel):
     array([-31.5871...])
     """
 
+    _parameter_constraints = {
+        "fit_intercept": ["boolean"],
+        "copy_X": ["boolean"],
+        "max_subpopulation": [Interval(Integral, 1, None, closed="left")],
```

Comment:
Based on the to-be-replaced `check_scalar` call, `max_subpopulation` should also accept floats for backwards compatability, hence you would want to include something like this:

```suggestion
        "max_subpopulation": [Interval(Integral, 1, None, closed="left"), Interval(Real, 1, None, closed="left")],
```

You'll have to fix the formatting accordingly though since this suggestion will probably make our linters unhappy.

Reply:
I'd even borrow the comment from the old validation to reduce confusion
```suggestion
        # target_type should be Integral but can accept Real for backward compatibility
        "max_subpopulation": [Interval(Integral, 1, None, closed="left"), Interval(Real, 1, None, closed="left")],
```